src/cljs/ataru/virkailija/editor/components/text_component.cljs

```diff
@@ -165,71 +184,88 @@
+         [:label.editor-form__text-field-option-condition-label
+          {:for   id
+           :class (when @component-locked? "editor-form__textfield-option-condition--disabled")}
+          @(subscribe [:editor/virkailija-translation :lisakysymys-arvon-perusteella-ehto])]
+         [:select.editor-form__text-field-option-condition-comparison-operator
+          {:disabled  @component-locked?
+           :on-change (fn [event]
+                        (dispatch [:editor/aseta-lisäkysymys-arvon-perusteella-operaattori
+                                   path
+                                   option-index
+                                   (get-val event)]))
+           :value     (or (-> condition :comparison-operator) "<")}
+          [:option {:value "<"} @(subscribe [:editor/virkailija-translation :lisakysymys-arvon-perusteella-ehto-pienempi])]
+          [:option {:value "="} @(subscribe [:editor/virkailija-translation :lisakysymys-arvon-perusteella-ehto-yhtasuuri])]
+          [:option {:value ">"} @(subscribe [:editor/virkailija-translation :lisakysymys-arvon-perusteella-ehto-suurempi])]]
+         [:input.editor-form__text-field-option-condition-answer-compared-to
+          {:disabled  @component-locked?
+           :class     (when (not (:valid? @local-state))
+                        "editor-form__text-field-option-condition-answer-compared-to--invalid")
+           :id        id
+           :on-blur   (partial on-blur-fn path)
+           :on-change on-change-fn
+           :type      "text"
+           :value     (:value @local-state)}]]))))
 
 (defn- text-field-option-followups-wrapper
-  [options followups path show-followups]
-  (let [option-count (count options)]
-    (when (or (nil? @show-followups)
-              (not (= (count @show-followups) option-count)))
-      (reset! show-followups (vec (replicate option-count true))))
-    (when (< 0 option-count)
-      (let [option-index 0
-            followups    (nth followups option-index)
-            option       (nth options option-index)]
-        [:div.editor-form__text-field-option-followups-wrapper
-         (when (:condition option)
-           [:div.editor-form__text-field-option-followups-header
-            [text-field-option-condition (:condition option) path]
-            [followup-question/followup-question option-index followups show-followups]])
-         [followup-question/followup-question-overlay option-index followups path show-followups]]))))
+  [_]
+  (let [show-followups (r/atom nil)]
+    (fn [{:keys [component-locked? followups options path]}]
+      (let [option-count                      (count options)
+            show-if-options-without-condition (not (empty? (remove :condition options)))]
+        (when (or (nil? @show-followups)
+                  (not (= (count @show-followups) option-count)))
+          (reset! show-followups (vec (replicate option-count show-if-options-without-condition))))
+        [:<>
+         (doall
```

Comment:
Tässä on aika monta sisennystasoa, joten voisi miettiä, saisiko sieltä ekstraktoitua jonun selkiyttävän funktion. Esimerkiksi `(when (:condition option)`:in alta voisi olla luonteva katkaista.